src/item/component/IconComponent.php

```diff
@@ -16,12 +18,14 @@ public function getName(): string {
 	}
 
 	public function getValue(): array {
-		return [
-			"texture" => $this->texture
-		];
+		return CompoundTag::create()
+			->setTag("textures",
+				CompoundTag::create()
+					->setString("default", $this->texture)
+			);
```

Comment:
This will definitely not work, the getValue return value is an array (forcely typed) so you cannot return a compoundtag

Reply:
```suggestion
		return [
			"textures" => [
				"default" => $this->texture
			]
		];
```
It will work with this, but not sure that's what you intend to do